lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java

```diff
@@ -85,7 +87,8 @@
                           final ClasspathProvider classpath,
                           final YarnProxyUser yarnProxyUser,
                           final SecurityTokenProvider tokenProvider,
-                          final YarnSubmissionParametersFileGenerator jobSubmissionParametersGenerator) {
+                          final YarnSubmissionParametersFileGenerator jobSubmissionParametersGenerator,
+                          @Parameter(NodeLabelExpression.class) final String nodeLabelExpression) {
```

Comment:
A user can label on the job submission to label entire evaluator requests for the job.
To label on the job, you can set node label on driverConfiguraion before run a driver.
```java
final Configuration driverConfiguration = DriverConfiguration.CONF
.set(NodeLabelTestConfiguration.NODE_LABEL_EXPRESSION, "mylabel")
...
.build();
```
`NODE_LABEL_EXPRESSION` is an optional parameter.

The change is for java only.